### DISCUSSION TITLE
Expose more NKeys / Ed25519 types / logic

### DISCUSSION THREAD

**Discussion by osigurdson** [nats-io/nats.net.v1#857] [General]
The NATS nsc tool is nice but can be hard to automate in an application. It would be great to be able to create new user jwts using this library. It appears that some of this already exists in namespace NATS.Client.Core.NaCl with internal types like Ed25519.

**Comment by scottf** [comment]
Is this what you are looking for?

https://github.com/nats-io/nats.net/blob/main/src/NATS.Client/NKeys.cs
https://github.com/nats-io/nats.net/blob/main/src/NATS.Client/Internals/JwtUtils.cs

We are also in the process of extracting both of these to their own repo, although with .NET, the extraction will target .NET versions that are used in the [.NET V2 client](https://github.com/nats-io/nats.net.v2)

Those new repos can be found at https://github.com/nats-io/nkeys.net and https://github.com/nats-io/jwt.net but currently they are empty.

The .NET V2 client does have this:
https://github.com/nats-io/nats.net.v2/blob/main/src/NATS.Client.Core/NKeyPair.cs
